msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -245,24 +313,38 @@ class AccountState private constructor(
             val acquireTokenSilentParameters = AcquireTokenSilentParameters.Builder()
                 .forAccount(account)
                 .fromAuthority(account.authority)
+                .forceRefresh(forceRefresh)
+                .withScopes(mergedScopes)
                 .build()
 
-            val accountToBeUsed = PublicClientApplication.selectAccountRecordForTokenRequest(
+            val accountRecord = PublicClientApplication.selectAccountRecordForTokenRequest(
                 config,
                 acquireTokenSilentParameters
             )
+            acquireTokenSilentParameters.accountRecord = accountRecord
 
-            val params = CommandParametersAdapter.createAcquireTokenNoFixedScopesCommandParameters(
+            val params = CommandParametersAdapter.createSilentTokenCommandParameters(
                 config,
                 config.oAuth2TokenCache,
-                accountToBeUsed,
-                forceRefresh,
-                correlationId
+                acquireTokenSilentParameters
             )
 
-            val command = AcquireTokenNoFixedScopesCommand(
+            val command = SilentTokenCommand(
                 params,
-                NativeAuthMsalController(),
+                NativeAuthMsalController().asControllerFactory(),
```

Comment:
This factory is a new pattern. Do we know why MSAL did this, and is this something we should include in native auth?

Reply:
Yes, Dome added this last month. We can update our code to use this method.

Reply:
Do we know why they did it? Without that I don't think we can determine whether we should include it.